action/protocol/execution/evm/evmstatedbadapter.go

```diff
@@ -540,6 +552,13 @@ func (stateDB *StateDBAdapter) RevertToSnapshot(snapshot int) {
 		log.L().Error("Failed to get snapshot.", zap.Int("snapshot", snapshot))
 		return
 	}
+	// restore gas refund
+	if !stateDB.notCorrectGasRefund {
```

Comment:
if stateDB.notCorrectGasRefund {} 
else {
}

Reply:
@Liuhaai  i also feel double negative is confused, i used `correctGasRefund` at the beginning

@CoderZhi can we use `correctGasRefund = true` for the default case (that is after hard-fork)? I know you would like it to be `false` but IMO it's not a big issue (as compared to use double negative and code readability). Let me know

Reply:
1. you misunderstand what @Liuhaai means...
2. `notCorrectGasRefund` could be renamed to `revertRefund`.